src/printer/smt2/smt2_printer.cpp

```diff
@@ -322,30 +322,42 @@ void Smt2Printer::toStream(std::ostream& out,
     return;
   }
 
-  if (!force_nt.isNull())
+  // determine if we are printing out a type ascription, store the argument of
+  // the type ascription into type_asc_arg.
+  Node type_asc_arg;
+  if (n.getKind() == kind::APPLY_TYPE_ASCRIPTION)
   {
-    if (n.getType() != force_nt)
+    force_nt = TypeNode::fromType(
+        n.getOperator().getConst<AscriptionType>().getType());
+    type_asc_arg = n[0];
+  }
+  else if (!force_nt.isNull() && n.getType() != force_nt)
+  {
+    type_asc_arg = n;
+  }
+  if (!type_asc_arg.isNull())
+  {
+    if (force_nt.isReal())
     {
-      if (force_nt.isReal())
-      {
-        out << "(" << smtKindString(force_nt.isInteger() ? kind::TO_INTEGER
-                                                         : kind::TO_REAL,
-                                    d_variant)
-            << " ";
-        toStream(out, n, toDepth, types, TypeNode::null());
-        out << ")";
-      }
-      else
-      {
-        Node nn = NodeManager::currentNM()->mkNode(
-            kind::APPLY_TYPE_ASCRIPTION,
-            NodeManager::currentNM()->mkConst(
-                AscriptionType(force_nt.toType())),
-            n);
-        toStream(out, nn, toDepth, types, TypeNode::null());
-      }
-      return;
+      out << "(" << smtKindString(
+                        force_nt.isInteger() ? kind::TO_INTEGER : kind::TO_REAL,
```

Comment:
Isn't it problematic that we output a to_int/to_real here? Shouldn't we use the division trick?

Reply:
The issue with ``/`` is that it is not allowed when n is not a constant, when the logic is linear.
It always works if we are NIA, NRA, or NIRA, and it always works if n is constant.

The issue with ``to_real`` as mentioned is that it is only allowed in the mixed Int/Real theory.
It always works if we are LIRA, or NIRA.

I think it may be better to do the division trick for now.